client/src/partials/patients/record/patient_record.js

```diff
@@ -1,26 +1,59 @@
 angular.module('bhima.controllers')
 .controller('PatientRecordController', PatientRecordController);
 
-PatientRecordController.$inject = ['$stateParams', 'PatientService', 'NotifyService', 'moment'];
+PatientRecordController.$inject = ['$stateParams', 'PatientService', 'NotifyService', 'moment', 'Upload', '$timeout', '$translate'];
 
-function PatientRecordController($stateParams, Patients, Notify, moment) {
+function PatientRecordController($stateParams, Patients, Notify, moment, Upload, $timeout, $translate) {
   var vm = this;
   var patientID = $stateParams.patientID;
   
   vm.loading = true;
+  vm.uploadFiles = uploadFiles;
+  vm.textButton = $translate.instant('PATIENT_RECORDS.UPLOAD_PICTURE');
+ 
+  refreshPatientsRead();
 
-  /** @fixme if no uuid is provided this will download all the patients through the base url '/' */
-  Patients.read(patientID)
-    .then(function (result) {
-      vm.patient = result;
-      vm.loading = false;
+  function uploadFiles(file, errFiles) {
+    vm.f = file; 
+    if (file) {
+      var imageCheck = file.type.search("image/");
+      if(imageCheck !== -1){
+        file.upload = Upload.upload({
+          url: '/patients/' + patientID + '/pictures', 
+          data: {pictures: file}
+        });
 
-      /** @todo move to service or mysql query */
-      vm.patient.name = [vm.patient.first_name, vm.patient.middle_name, vm.patient.last_name].join(' ');
-      vm.patient.age = moment().diff(vm.patient.dob, 'years');
-    })
-    .catch(function (error) {
-      vm.loading = false;
-      Notify.handleError(error);
-    });
-}
+        file.upload.then(function (response) {
```

Comment:
This function is missing a `.catch()` statement.  I think there are some errors that are silently failing because of this.

Could you add that?  It is probably the cause of https://github.com/IMA-WorldHealth/bhima-2.X/pull/529#issuecomment-228290716.